whylog/tests/tests_config/test_concated_regex.py

```diff
@@ -0,0 +1,240 @@
+        regex2 = "^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) Data is missing at (.*)$"
+        regex3 = "^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) Connection error occurred on (.*)\. Host name: (.*)$"
+        regex4 = "^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) Data migration from (.*) to (.*) failed\. Host name: (.*)$"
+
+        parser_intent1 = UserParserIntent("date", "filesystem", regex1, [1], {1: to_date})
+        parser_intent2 = UserParserIntent("onlymissdata", "filesystem", regex2, [1], {1: to_date})
+        parser_intent3 = UserParserIntent("connectionerror", "hydra", regex3, [1], {1: to_date})
+        parser_intent4 = UserParserIntent("datamigration", "hydra", regex4, [1], {1: to_date})
+
+        parser1 = RegexParserFactory.create_from_intent(parser_intent1)
+        parser2 = RegexParserFactory.create_from_intent(parser_intent2)
+        parser3 = RegexParserFactory.create_from_intent(parser_intent3)
+        parser4 = RegexParserFactory.create_from_intent(parser_intent4)
+
+        content = "2015-12-03 12:11:00 Data is missing at alfa21. Loss = 567.02 GB. Host name: 101"
+
+        concated = ConcatedRegexParser([parser1, parser2, parser3, parser4])
+
+        assert concated.get_extracted_regex_params(content) == {
+            parser1.name: ["2015-12-03 12:11:00", "alfa21", "567.02", "101"],
+            parser2.name: ["2015-12-03 12:11:00", "alfa21. Loss = 567.02 GB. Host name: 101"],
+        }
+
+        concated = ConcatedRegexParser([parser4, parser3, parser2, parser1])
+
+        assert concated.get_extracted_regex_params(content) == {
+            parser1.name: ["2015-12-03 12:11:00", "alfa21", "567.02", "101"],
+            parser2.name: ["2015-12-03 12:11:00", "alfa21. Loss = 567.02 GB. Host name: 101"],
+        }
+
+    def test_single_subregex(self):
+        regex1 = "^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) Data is missing at (.*)\. Loss = (.*) GB\. Host name: (.*)$"
+        regex2 = "^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) Data is missing at (.*)$"
+
+        parser_intent1 = UserParserIntent("lostdata", "filesystem", regex1, [1], {1: to_date})
+        parser_intent2 = UserParserIntent("onlymissdata", "filesystem", regex2, [1], {1: to_date})
+
+        parser1 = RegexParserFactory.create_from_intent(parser_intent1)
+        parser2 = RegexParserFactory.create_from_intent(parser_intent2)
+
+        content = "2015-12-03 12:11:00 Data is missing at alfa21. Loss = 567.02 GB. Host name: 101"
+
+        concated = ConcatedRegexParser([parser1])
+
+        assert concated.get_extracted_regex_params(content) == {
+            parser1.name: ["2015-12-03 12:11:00", "alfa21", "567.02", "101"],
+        }
+
+        concated = ConcatedRegexParser([parser2])
+
+        assert concated.get_extracted_regex_params(content) == {
+            parser2.name: ["2015-12-03 12:11:00", "alfa21. Loss = 567.02 GB. Host name: 101"],
+        }
+
+    def test_large_concated_regex(self):
```

Comment:
this "test" is in fact many testcases which live in a single test method. Please refactor so that there are much more methods. You can put most of the code in a helper method which will really do all the work, but then each test will have a name and purpose and the code will not need to be copied.